File: frontend/src/models/AppointmentDTO.ts
import PricingDTO from "./PricingDTO";
import UserDTO from "./UserDTO";

class AppointmentDTO{
    id?: number;
    date: string;
    status: string;
    patient: UserDTO;
    doctor: UserDTO;
    pricing: PricingDTO;
    paid: boolean;

    constructor(date: string, status: string, patient: UserDTO, doctor: UserDTO, pricing: PricingDTO, paid: boolean, id?: number){
        this.date = date;
        this.status = status;
        this.patient = patient;
        this.doctor = doctor;
        this.pricing = pricing;
        this.paid = paid;
        this.id = id;
    }
}

export default AppointmentDTO;

File: backend/src/main/java/com/serhiienko/backend/model/dto/AppointmentDTO.java
package com.serhiienko.backend.model.dto;

import com.serhiienko.backend.model.entity.Pricing;
import com.serhiienko.backend.model.enumeration.AppointmentStatus;
import com.serhiienko.backend.model.enumeration.EstimatedTime;
import lombok.Builder;
import lombok.Data;

import java.util.Date;

@Builder
@Data
public class AppointmentDTO {
    private Long id;
    private Date date;
    private AppointmentStatus status;
    private UserDTO patient;
    private UserDTO doctor;
    private Pricing pricing;
    private boolean paid;
}
